community/neo4j-harness/src/test/java/org/neo4j/harness/JavaProceduresTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.harness;
+
+import org.codehaus.jackson.JsonNode;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.stream.Stream;
+
+import org.neo4j.proc.ReadOnlyProcedure;
+import org.neo4j.test.SuppressOutput;
+import org.neo4j.test.TargetDirectory;
+import org.neo4j.test.server.HTTP;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.neo4j.test.server.HTTP.RawPayload.quotedJson;
+
+public class JavaProceduresTest
```

Comment:
Perhaps some negative tests as well? What's intended the response if

* procedure throws an exception? (Empty data atm)
* procedure doesn't exist (Null result atm)